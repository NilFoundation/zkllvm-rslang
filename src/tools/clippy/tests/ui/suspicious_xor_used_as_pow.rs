#![allow(unused)]
#![warn(clippy::suspicious_xor_used_as_pow)]
#![allow(clippy::eq_op)]

macro_rules! macro_test {
    () => {
        13
    };
}

macro_rules! macro_test_inside {
    () => {
        1 ^ 2 // should warn even if inside macro
    };
}

fn main() {
    // Should warn:
    let _ = 2 ^ 5;
    let _ = 2i32 ^ 9i32;
    let _ = 2i32 ^ 2i32;
    let _ = 50i32 ^ 3i32;
    let _ = 5i32 ^ 8i32;
    let _ = 2i32 ^ 32i32;
    macro_test_inside!();

    // Should not warn:
    let x = 0x02;
    let _ = x ^ 2;
    let _ = 2 ^ x;
    let _ = x ^ 5;
    let _ = 10 ^ 0b0101;
    let _ = 2i32 ^ macro_test!();
}
