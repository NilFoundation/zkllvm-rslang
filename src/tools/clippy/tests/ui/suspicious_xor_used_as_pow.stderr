error: `^` is not the exponentiation operator
  --> $DIR/suspicious_xor_used_as_pow.rs:19:13
   |
LL |     let _ = 2 ^ 5;
   |             ^^^^^ help: did you mean to write: `2.pow(5)`
   |
   = note: `-D clippy::suspicious-xor-used-as-pow` implied by `-D warnings`

error: `^` is not the exponentiation operator
  --> $DIR/suspicious_xor_used_as_pow.rs:20:13
   |
LL |     let _ = 2i32 ^ 9i32;
   |             ^^^^^^^^^^^ help: did you mean to write: `2_i32.pow(9_i32)`

error: `^` is not the exponentiation operator
  --> $DIR/suspicious_xor_used_as_pow.rs:21:13
   |
LL |     let _ = 2i32 ^ 2i32;
   |             ^^^^^^^^^^^ help: did you mean to write: `2_i32.pow(2_i32)`

error: `^` is not the exponentiation operator
  --> $DIR/suspicious_xor_used_as_pow.rs:22:13
   |
LL |     let _ = 50i32 ^ 3i32;
   |             ^^^^^^^^^^^^ help: did you mean to write: `50_i32.pow(3_i32)`

error: `^` is not the exponentiation operator
  --> $DIR/suspicious_xor_used_as_pow.rs:23:13
   |
LL |     let _ = 5i32 ^ 8i32;
   |             ^^^^^^^^^^^ help: did you mean to write: `5_i32.pow(8_i32)`

error: `^` is not the exponentiation operator
  --> $DIR/suspicious_xor_used_as_pow.rs:24:13
   |
LL |     let _ = 2i32 ^ 32i32;
   |             ^^^^^^^^^^^^ help: did you mean to write: `2_i32.pow(32_i32)`

error: `^` is not the exponentiation operator
  --> $DIR/suspicious_xor_used_as_pow.rs:13:9
   |
LL |         1 ^ 2 // should warn even if inside macro
   |         ^^^^^ help: did you mean to write: `1.pow(2)`
...
LL |     macro_test_inside!();
   |     -------------------- in this macro invocation
   |
   = note: this error originates in the macro `macro_test_inside` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 7 previous errors

